code/modules/mob/living/carbon/life.dm

```diff
@@ -602,14 +602,14 @@ GLOBAL_LIST_INIT(ballmer_windows_me_msg, list("Yo man, what if, we like, uh, put
 
 /mob/living/carbon/proc/handle_bodycheck()
 	for(var/mob/H in oview(src, 7))
-		if(H.stat == DEAD)
-			for(var/F in src.faction)
-				if(F in H.faction)
-					if(F != "neutral") //doesn't work merged with above, fix if you can
-						if(LAZYFIND(src.faction_deaths, H.real_name)) //same here
-							return
-						else
-							LAZYADD(src.faction_deaths, H.real_name)
-							to_chat(src, src.faction_deaths.len)
-							if(LAZYLEN(src.faction_deaths) >= 3)
-								SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "saw_many_unburied_faction", /datum/mood_event/saw_many_unburied_faction)
\ No newline at end of file
+		if(H.stat != DEAD || !istype(H)) //fuck off
+			return 
+		for(var/F in faction)
+			if(F in H.faction)
+				if(F == "neutral" || LAZYFIND(faction_deaths, H.real_name)) //same here
+					return
```

Comment:
Instead of this you should check first if the source is from a neutral faction and do an early return if it's the case.